pkg/registry/rbac/validation/rule_test.go

```diff
@@ -275,3 +283,121 @@ func TestAppliesTo(t *testing.T) {
 		}
 	}
 }
+
+type conditionalAttributes struct {
+	namespace         string
+	name              string
+	user              user.DefaultInfo
+	isResourceRequest bool
+	fieldSelector     fields.Requirements
+	labelSelector     labels.Requirements
+}
+
+var _ cel.ConditionalAttributes = &conditionalAttributes{}
+
+func (c *conditionalAttributes) GetNamespace() string {
+	return c.namespace
+}
+
+func (c *conditionalAttributes) GetName() string {
+	return c.name
+}
+
+func (c *conditionalAttributes) GetUser() user.Info {
+	return &c.user
+}
+
+func (c *conditionalAttributes) IsResourceRequest() bool {
+	return c.isResourceRequest
+}
+
+func (c *conditionalAttributes) GetFieldSelector() (fields.Requirements, error) {
+	return c.fieldSelector, nil
+}
+
+func (c *conditionalAttributes) GetLabelSelector() (labels.Requirements, error) {
+	return c.labelSelector, nil
+}
+
+func labelRequirement(t *testing.T, key string, operator selection.Operator, values []string) labels.Requirement {
+	req, err := labels.NewRequirement(key, operator, values)
+	if err != nil {
+		t.Fatalf("error creating label requirement: %v", err)
+	}
+	return *req
+}
+
+func TestConditionalClusterRoleBindingAppliesTo(t *testing.T) {
+	tests := []struct {
+		name                          string
+		conditionalClusterRoleBinding *rbacv1alpha1.ConditionalClusterRoleBinding
+		conditionalAttributes         *conditionalAttributes
+		expected                      bool
+	}{
+		{
+			name:                          "no conditions",
+			conditionalClusterRoleBinding: &rbacv1alpha1.ConditionalClusterRoleBinding{},
+			expected:                      false,
+		},
+		{
+			name: "listing resource with specific label",
+			conditionalClusterRoleBinding: &rbacv1alpha1.ConditionalClusterRoleBinding{
+				Conditions: []rbacv1alpha1.Condition{
+					{
+						Expression: "request.user == 'foobar'",
+					},
+					{
+						Expression: "request.resourceAttributes.labelSelector.requirements.exists(r, r.key == 'hellokey' && r.operator == '=' && ('valuefoo' in r.values))",
+					},
+				},
+			},
+			conditionalAttributes: &conditionalAttributes{
+				user: user.DefaultInfo{Name: "foobar"},
+				labelSelector: labels.Requirements{
+					labelRequirement(t, "hellokey", selection.Equals, []string{"valuefoo"}),
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "node operation restricted based on field selector",
+			conditionalClusterRoleBinding: &rbacv1alpha1.ConditionalClusterRoleBinding{
+				Conditions: []rbacv1alpha1.Condition{
+					{
+						// determine it's a node by checking the prefix and asserting the name is not just the prefix
+						Expression: "request.user.startsWith('system:node') && request.user != 'system:node'",
```

Comment:
btw I think `!=` violates the no negative rule?

```suggestion
						Expression: "request.user.startsWith('system:node:') && request.user != 'system:node:'",
```

Reply:
changed it to check the length instead